spot_driver/spot_driver/spot_ros2.py

```diff
@@ -445,87 +456,128 @@ def _goal_complete(self, feedback):
-                  feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                          FEEDBACK_STANCE_FEEDBACK_SET):
-                if (feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                        stance_feedback.status.value ==
-                        feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                                stance_feedback.status.STATUS_STANCED):
-                    return True
-                if (feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                        stance_feedback.status.value ==
-                        feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                                stance_feedback.status.STATUS_TOO_FAR_AWAY):
-                    return -1
-                return False
-            elif (feedback.command.synchronized_feedback.mobility_command_feedback.feedback.feedback_choice ==
-                  feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                          FEEDBACK_STOP_FEEDBACK_SET):
-                return True
-            elif (feedback.command.synchronized_feedback.mobility_command_feedback.feedback.feedback_choice ==
-                  feedback.command.synchronized_feedback.mobility_command_feedback.feedback.
-                          FEEDBACK_FOLLOW_ARM_FEEDBACK_SET):
-                return True
+                    return IN_PROGRESS
+                return FAILED
             else:
-                return False
-        return False
+                return IN_PROGRESS
+
+        elif feedback.command.command_choice == feedback.command.COMMAND_SYNCHRONIZED_FEEDBACK_SET:
+            # The idea here is that a synchronized command can have arm, mobility, and/or gripper
+            # sub-commands in it.  The total command isn't done until all sub-commands are satisfied.
+            # So if any one of the sub-commands is still in progress, it short-circuits out as
+            # IN_PROGRESS.  And if it makes it to the bottom of the function, then all components
+            # must be satisfied, and it returns SUCCESS.
+            # One corner case to know about is that the commands that don't provide feedback, such
+            # as a velocity command will return SUCCESS.  This allows you to use them more effectively
+            # with other commands.  For example if you wanted to move the arm with some velocity
+            # while the mobility is going to some SE2 trajectory then that will work.
+
+            sync_feedback = feedback.command.synchronized_feedback
+            if sync_feedback.arm_command_feedback_is_set == True:
+                arm_feedback =  sync_feedback.arm_command_feedback
+                if (arm_feedback.status.value != arm_feedback.status.STATUS_PROCESSING):
+                    return IN_PROGRESS
```

Comment:
this could probably use a comment because it's counter-intuitive

Reply:
I agree.  I think if this is anything other than `STATUS_PROCESSING` then we should probably return `FAIL` instead of `IN_PROGRESS`.  I think I was afraid that it might return `STATUS_UNKNOWN` for an iteration before `STATUS_PROCESSING`.  I'm a little new to the spot_driver, so I was also just following what the previous version did.

For reference, the other values in the enum are:
```
#/ Behavior execution is in an unknown / unexpected state.
int8 STATUS_UNKNOWN=0

#/ The robot is actively working on the command
int8 STATUS_PROCESSING=1

#/ The command was replaced by a new command
int8 STATUS_COMMAND_OVERRIDDEN=2

#/ The command expired
int8 STATUS_COMMAND_TIMED_OUT=3

#/ The robot is in an unsafe state, and will only respond to known safe commands.
int8 STATUS_ROBOT_FROZEN=4

#/ The request cannot be executed because the required hardware is missing.
#/ For example, an armless robot receiving a synchronized command with an arm_command
#/ request will return this value in the arm_command_feedback status.
int8 STATUS_INCOMPATIBLE_HARDWARE=5
```

@jbarry-bdai If you think it should actually return `FAIL`, let me know, it is an easy change.

Reply:
I think it returns `STATUS_UNKNOWN` right at the beginning before it goes into processing.